### DISCUSSION TITLE
Middleware always returns 403 when using  $this->actingAs($user, 'api') in unit tests.

### DISCUSSION THREAD

**Discussion by sprobejames** [spatie/laravel-permission#2603] [Q&A]
The `$user = $authGuard->user();` in **Spatie\Permission\Middleware\RoleMiddleware** is always `null` when using `$this->actingAs($user, 'api')->json('GET', '/some-route');` in feature testing. However, if the url is tested directly using postman with bearer token, it works. 

Is there a way to test role/permission in route middleware using the `$this->actingAs($user, api)` method?

**Comment by parallels999** [comment]
Try changing default guard
```php
config(['auth.default.guard' => 'api'])
``` 
Or set a testing bearer token
https://github.com/spatie/laravel-permission/blob/0e593bb9e6808cc55c67f150878b0847f20b9036/tests/TestCase.php#L276

**Reply by parallels999** [reply]
https://github.com/spatie/laravel-permission/pull/2556#issuecomment-1814478878
If you are using Passport
```php
Passport::actingAs($user);
```
https://github.com/spatie/laravel-permission/pull/2557
